kotsadm/pkg/kurl/join_cert.go

```diff
@@ -0,0 +1,199 @@
+package kurl
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/kotsadm/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func createCertAndKey(ctx context.Context, client kubernetes.Interface, namespace string) (string, error) {
+	pod, err := getPodSpec(client, namespace)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create pod spec")
+	}
+
+	_, err = client.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "failed to create pod")
+	}
+
+	defer func() {
+		go func() {
+			if err := client.CoreV1().Pods(pod.Namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{}); err != nil {
+				logger.Errorf("Failed to delete pod %s: %v\n", pod.Name, err)
+			}
+		}()
+	}()
+
+	for {
+		status, err := client.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", errors.Wrap(err, "failed to get pod")
+		}
+		if status.Status.Phase == corev1.PodRunning ||
+			status.Status.Phase == corev1.PodFailed ||
+			status.Status.Phase == corev1.PodSucceeded {
+			break
+		}
+		time.Sleep(time.Second * 1)
```

Comment:
this could hang forever, wdyt about adding a timeout in here?

Reply:
better yet use the context with timeout

```go
if err := ctx.Err(); err != nil {
  return err
}
```